data_pipeline/README.md

```diff
@@ -30,14 +29,69 @@ We will publish the code to generate the speed classifications from the traffic
+            if len(volumes_1h) != 5:
+                continue
+            r['t'] = t
+            r['volumes_1h'] = volumes_1h[:-1]
+            result.append(r.copy())
+            volumes_1h = volumes_1h[1:]
+    return pandas.DataFrame(result)
+
+def provision_loop_counter_input(city, month, days):
+    input_parquet = f'loop_counters/<CITY>/counters_normalized_{month}.parquet'
+    input_df = pandas.read_parquet(input_parquet)
+    print(f'Read {len(input_df)} rows from {input_parquet}')
+    input_df = input_df.reset_index()
+    input_df['node_id'] = input_df['node_id'].astype('int64')
+    input_df = input_df.groupby(by=['node_id']).apply(combine_1h_volume)
+    input_df = input_df.reset_index()
+    input_df = input_df[['node_id', 'day', 't', 'volumes_1h']]
+    for day in days:
+        provision_df = input_df[(input_df["day"]==day)]
+        print(f' Result rows: {len(provision_df)}')
+        print(f' Counters: {len(provision_df["node_id"].unique())}')
+        provision_parquet = f'train/{city}/input/counters_{day}.parquet'
+        provision_df.to_parquet(provision_parquet, compression='snappy')
+        print(f'Wrote training input to {provision_parquet}')
+```
+
+## Dynamic Edge Data: Congestion Classes from GPS Speeds
 <img src="../img/data_pipeline_edge_data.svg">
 
+The speed class data from MeTS-10 is compatible with the Traffic4cast 2022 process for CC labels: generate congestion classes (green=1, yellow=2, red=3) from the current segment medium speeds, the free flow speeds computed for the segment from the traffic map movies and the OSM signalled speeds. If no or not enough dynamic speed data is available, do not classify (unclassified=0).
+
+Use the scripts from https://github.com/iarai/MeTS-10/tree/master/data_pipeline after running steps 1-3 above:
+* `dp04_intersecting_cells.py`
+* `dp05_free_flow.py`
+* `dp06_speed_classes.py`
+
+Run `dp05_free_flow.py`  with the following custom arguments:
+* `--use_speed_limit`
+
+Outputs of `dp06_speed_classes.py` will be stored as `speed_classes_<date>.parquet` to `speed_classes/<city>`.
 
-| Item               | Description                |
-|--------------------|----------------------------|
-| Road Selection     | Select OSM primary features of type "highway" in the bounding box, introduce nodes for loop counters and simplify the road graph (remove dead-ends and unconnected components etc.). |
-| Spatial Join       | Intersect OSM roads geometries with map movie cell geometries. |
-| Combination        | Combine Traffic Map values with OSM IDs. |
-| Generate CC Labels | Generate congestion classes (green=1, yellow=2, red=3) from the current segment medium speeds, the free flow speeds computed for the segment from the traffic map movies and the OSM signalled speeds. If no or not enough dynamic speed data is available, do not classify (unclassified=0). |
+With these outputs `t4c22/prepare_training_data_cc.py` can be used to generate the training labels.
```

Comment:
Do we want to publish super-segment sampling code as well? (see (CODE-3) in overleaf)

Reply:
TODO